packages/core/src/jsx-runtime.js

```diff
@@ -0,0 +1,174 @@
+
+const sanitizeIdentifier = identifier => identifier.replace(/\$/g, '-')
+
+let typePropName = '__EMOTION_TYPE_PLEASE_DO_NOT_USE__'
+
+let labelPropName = '__EMOTION_LABEL_PLEASE_DO_NOT_USE__'
+
+let hasOwnProperty = Object.prototype.hasOwnProperty
+
+let render = (cache, props, theme, ref) => {
+  let cssProp = theme === null ? props.css : props.css(theme)
+
+  // so that using `css` from `emotion` and passing the result to the css prop works
+  // not passing the registered cache to serializeStyles because it would
+  // make certain babel optimisations not possible
+  if (typeof cssProp === 'string' && cache.registered[cssProp] !== undefined) {
+    cssProp = cache.registered[cssProp]
+  }
+
+  let type = props[typePropName]
+  let registeredStyles = [cssProp]
+  let className = ''
+
+  if (typeof props.className === 'string') {
+    className = getRegisteredStyles(
+      cache.registered,
+      registeredStyles,
+      props.className
+    )
+  } else if (props.className != null) {
+    className = `${props.className} `
+  }
+
+  let serialized = serializeStyles(registeredStyles)
+
+  if (
+    process.env.NODE_ENV !== 'production' &&
+    serialized.name.indexOf('-') === -1
+  ) {
+    let labelFromStack = props[labelPropName]
+    if (labelFromStack) {
+      serialized = serializeStyles([
+        serialized,
+        'label:' + labelFromStack + ';'
+      ])
+    }
+  }
+  const rules = insertStyles(cache, serialized, typeof type === 'string')
+  className += `${cache.key}-${serialized.name}`
+
+  const newProps = {}
+  for (let key in props) {
+    if (
+      hasOwnProperty.call(props, key) &&
+      key !== 'css' &&
+      key !== typePropName &&
+      (process.env.NODE_ENV === 'production' || key !== labelPropName)
+    ) {
+      newProps[key] = props[key]
+    }
+  }
+  newProps.ref = ref
+  newProps.className = className
+
+  const ele = jsx(type, newProps)
+  if (!isBrowser && rules !== undefined) {
+    let serializedNames = serialized.name
+    let next = serialized.next
+    while (next !== undefined) {
+      serializedNames += ' ' + next.name
+      next = next.next
+    }
+    return (
+      <React.Fragment>
+        <style
+          {...{
+            [`data-emotion-${cache.key}`]: serializedNames,
+            dangerouslySetInnerHTML: { __html: rules },
+            nonce: cache.sheet.nonce
+          }}
+        />
+        {ele}
+      </React.Fragment>
+    )
+  }
+  return ele
+}
+
+let Emotion = /* #__PURE__ */ withEmotionCache((props, cache, ref) => {
+  // use Context.read for the theme when it's stable
+  if (typeof props.css === 'function') {
+    return (
+      <ThemeContext.Consumer>
+        {theme => render(cache, props, theme, ref)}
+      </ThemeContext.Consumer>
+    )
+  }
+  return render(cache, props, null, ref)
+})
+
+if (process.env.NODE_ENV !== 'production') {
+  Emotion.displayName = 'EmotionCssPropInternal'
+}
+
+const jsx = (type, props, maybeKey) => {
```

Comment:
change between this and old `jsx` is that it's called w/ a props object that includes children *without` `key` and passes key as the third arg. 

Gigantic caveat: I don't really grok this code very well yet, but I was able to get it sort of working. There's definitely something broken with keys.